migrations/sql/V2019.06.26.10.30__add_code_required_report_tables.sql

```diff
@@ -0,0 +1,154 @@
+    update_timestamp                     timestamp with time zone DEFAULT now()  NOT NULL   ,
+
+    FOREIGN KEY (compliance_article_id) REFERENCES compliance_article(compliance_article_id) DEFERRABLE INITIALLY DEFERRED,
+    FOREIGN KEY (mine_report_due_date_type) REFERENCES mine_report_due_date_type(mine_report_due_date_type) DEFERRABLE INITIALLY DEFERRED
+);
+
+COMMENT ON TABLE mine_report_definition is 'Definitions of code required reports, due dates and articles of the code that apply to them.';
+ALTER TABLE mine_report_definition OWNER TO mds;
+
+CREATE TABLE IF NOT EXISTS mine_report
+(
+    mine_report_id               serial                                   NOT NULL PRIMARY KEY,
+    mine_report_guid             uuid                                     NOT NULL            ,
+    mine_report_definition_id    integer                                  NOT NULL            ,
+    mine_guid                    uuid                                     NOT NULL            ,
+    permit_id                    integer                                                      ,
+    due_date                     timestamp                                NOT NULL            ,
+    submission_year              integer                                                      ,
+    deleted_ind                  boolean DEFAULT false                    NOT NULL            ,
+    create_user                  character varying(60)                    NOT NULL            ,
+    create_timestamp             timestamp with time zone DEFAULT now()   NOT NULL            ,
+    update_user                  character varying(60)                    NOT NULL            ,
+    update_timestamp             timestamp with time zone DEFAULT now()   NOT NULL            ,
+    
+    FOREIGN KEY (mine_report_definition_id) REFERENCES mine_report_definition(mine_report_definition_id) DEFERRABLE INITIALLY DEFERRED,
+    FOREIGN KEY (mine_guid) REFERENCES mine(mine_guid) DEFERRABLE INITIALLY DEFERRED,
+    FOREIGN KEY (permit_id) REFERENCES permit(permit_id) DEFERRABLE INITIALLY DEFERRED
+
+);
+
+COMMENT ON TABLE mine_report is 'A code required report for a specific mine or permit on a mine.';
+ALTER TABLE mine_report OWNER TO mds;
+
+CREATE TABLE IF NOT EXISTS mine_report_submission (
+    mine_report_submission_id             serial                                  NOT NULL PRIMARY KEY,
+    mine_report_submission_guid           uuid                                    NOT NULL            ,
+    mine_report_id                        integer                                 NOT NULL            ,
+    received_date                         timestamp                                                   ,
+    mine_report_submission_status_code    varchar(3)                              NOT NULL            ,
+    active_ind                            boolean DEFAULT true                    NOT NULL            ,
```

Comment:
Not sure I understand the use of an "active" submission. Can a submission expire? Or is this covering a soft-delete of a report that has been submitted?

Reply:
should this be deleted_ind?

Reply:
The logic for this is that there can be several submissions for a report and only the most recent is the active submission. You don't want to lose the history or the previous submissions though.

Reply:
Could this be managed through the submission_status_code?

Reply:
This was removed and planned to be handled through submission date.